### DISCUSSION TITLE
v0.9.0-preview1

### DISCUSSION THREAD

**Discussion by Blinue** [Blinue/Magpie#361] [Announcements]
**新功能**

1. 全面重写渲染系统，新的 MagpieFX v2，语法见[自定义效果](https://github.com/Blinue/Magpie/blob/dev/docs/%E8%87%AA%E5%AE%9A%E4%B9%89%E6%95%88%E6%9E%9C%EF%BC%88MagpieFX%EF%BC%89.md)
2. 实现游戏内覆盖，目前仅用于衡量性能，以后也会用于在运行时更改效果参数。默认快捷键 Ctrl+Alt+D，可在设置中更改。
![image](https://user-images.githubusercontent.com/34770031/167125685-5ed80262-6722-41da-87c1-a43b594f8396.png)
3. 拖拽 FPS 可调整位置，在 FPS 上右键可调整样式
4. 增强了光标功能，现在可以将光标自然的移到其他窗口上，如开始菜单、XBox Game Bar 和 Alt+Tab 窗口
5. 在 ScaleModels.json 中可以为效果指定 inlineParams 参数以启用内联参数。这可以稍微提高某些效果的性能，但每次更改参数时都需要重新编译该效果。

示例：
``` json
{
    "effect": "FSR_RCAS",
    "sharpness": 0.87,
    "inlineParams": true
}
```
6.  “在 3D 游戏中限制光标”已升级为“3D 游戏模式”，启用时将针对 3D 游戏进行优化，游戏内覆盖也有不同的表现。
7. 调整缩放模式预设，移除 RAVU。

**效果更改**

1. 所有效果均已重写。性能对比见 https://github.com/Blinue/Magpie/discussions/254#discussioncomment-2251071
2. 添加 CatmullRom，这是 Bicubic 的一个变体，针对速度优化，最适合下采样
3. 添加 CAS_Scaling，支持缩放的 CAS 变体
4. 添加 LCAS，一个非常轻量的 3D 画面缩放算法（感谢 @Tzugimaa）
5. 删除 Anime4K_Upscale_S_Lite、Bicubic_Lite 和 FSR_EASU_DX10
6. Linear 重命名为 Bilinear
8. AdaptiveSharpen 的 curveHeight 参数默认值调整为 0.8
9. 移除了 CRT_Easymode 的 scaleInLinearGamma 参数
10. SSimDownscaler 添加 oversharp 参数用于调整锐度

**系统需求**
1. Windows 10/11
2. **DirectX Feature Level 11**

<hr /><em>This discussion was created from the release <a href='https://github.com/Blinue/Magpie/releases/tag/v0.9-preview1'>v0.9.0-preview1</a>.</em>

**Comment by hooke007** [comment]
LCAS还是旧语法无法使用

**Reply by Blinue** [reply]
好吧。发布正式版之前会合并

**Comment by hooke007** [comment]
inlineParams 有没有什么差异较大的例子，我测了几个单项，从帧率和计时来看都没有较明显的区别

**Reply by Blinue** [reply]
我的测试里差异也非常小，CRT_Easymode 性能提高最多

**Comment by chaos700** [comment]
> 版本	Windows 11 專業版
> 版本	21H2
> OS 組建	22000.652
> 體驗	Windows 功能體驗套件 1000.22000.652.0

 点击exe后，没有启动程式，
说起来，之前v0.8.2也是不能启动，
后来用[#359](https://github.com/Blinue/Magpie/issues/359)里面的替代版本才能正常启动。

**Reply by Blinue** [reply]
进入事件查看器->windows日志->应用程序，看系统有记录到错误吗？

**Reply by chaos700** [reply]
自行发现问题原因，
电脑上的.NET 6原先版本是6.0.1，
更新到最新的6.0.4就能正常启动。

**Reply by Blinue** [reply]
小版本都不兼容，微软太不走心了

**Reply by hooke007** [reply]
<s>有点怪，我的本机sdk是6.0.201是正常启动的</s> 原来是vs自动更新把我手动装的覆盖了，实际版本高于602

**Reply by Blinue** [reply]
dotnet 的 SDK 版本和运行时版本有不同的版本号。SDK 目前是 6.0.202，对应 Runtime 是 6.0.4。

为了防止再出现不兼容的问题，我在 Actions 里把 SDK 的版本号锁定为 202，以后用户无需再更新 Runtime 了。

**Comment by Blinue** [comment]
事实上除了 inlineParams 还添加了 fp16 参数，但在我的测试里半精度并没有提高性能。目前只有 NIS 和 CAS 适配了半精度，在其他效果上都不起作用。

**Reply by MindSerene** [reply]
> 事实上除了 inlineParams 还添加了 fp16 参数，但在我的测试里半精度并没有提高性能。目前只有 NIS 和 CAS 适配了半精度，在其他效果上都不起作用。

Magpie中似乎没有NIS,那两个x2,x3好像只是单纯的缩放, 小黄鸭里有.相对FSR1.0来说CPU占用率更低,帧率更高,效果也更好.

**Reply by hooke007** [reply]
magpie的支持算法远多于LS，预设只给了几个例子。其它算法阅读维基的说明

**Reply by hooke007** [reply]
而且首页也有写 “👉 自定义缩放配置（示例）”

**Comment by TredZes** [comment]
Small issue: version 0.8.100 is displayed in the "About" window.

**Reply by Blinue** [reply]
This is by design.
